### PR TITLE
Closes #2385 Add indent and outdent to az_standard text editor

### LINKED ISSUE #2385: Add indent and outdent to az_standard text formatter.
## Motivation
Is it possible to add buttons for Indent and Outdent? There's a Quote, but that of course isn't quite the same. Thoughts? Thanks!

We need to determine how this will display for each breakpoint.

### PR DESCRIPTION
**Pull request by trackleft**
## Related issues
Closes #2385

Turns out, in order to enable indenting of content that is not a list, you need to download and install a contrib module.

https://www.drupal.org/project/ckeditor_indentblock

## How to test
Edit any content that uses the az_standard text formatter.
See two new buttons in the alignment section for indent and outdent.

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [x] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: az_quickstart.install [hunk 1]
```diff
@@ -281,3 +281,10 @@ function az_quickstart_update_920701() {
 function az_quickstart_update_920702() {
   \Drupal::service('module_installer')->install(['az_paragraphs_splitscreen']);
 }
+
+/**
+ * Enable az_paragraphs_splitscreen module by default.
```
Comment (trackleft) [inline az_quickstart.install:286]:
```suggestion
 * Enable ckeditor_indentblock module by default.
```

File: modules/custom/az_paragraphs/config/optional/editor.editor.az_standard.yml [hunk 2]
```diff
@@ -142,7 +144,6 @@ settings:
         - '<a class target name id title>'
         - '<drupal-entity data-entity-type data-embed-button alt title data-align data-caption data-entity-embed-display data-entity-embed-display-settings data-view-mode data-entity-uuid data-langcode>'
         - '<drupal-media title>'
```
Comment (trackleft) [inline modules/custom/az_paragraphs/config/optional/editor.editor.az_standard.yml:146]:
```suggestion
        - '<drupal-media title>'
        - '<p class>'
```

**Comment by trackleft** [comment]
When saving the az_standard editor config I see this error 
image
```
Error message
The following attribute(s) are already supported by enabled plugins and should not be added to the Source Editing "Manually editable HTML tags" field: Indent block (<p class>).
```
Until I remove `<p class>` from the "Manually editable HTML tags" field.
image

This effectively breaks the Source editing of paragraph classes function, so I opted to leave `<p class>` in that field by editing the config manually.

File: composer.json [hunk 3]
```diff
@@ -50,8 +50,8 @@
         "drupal/bootstrap_utilities": "2.0.0",
         "drupal/cas": "2.2.0",
         "drupal/chosen": "4.0.0",
-        "drupal/ckeditor": "*",
```
Comment (bberndt-uaz) [inline composer.json:53]: Is this an intended change?

**Review by ejsamboy — DISMISSED** [review]
Screenshot 2024-02-21 at 9 54 30 AM

**Review by bberndt-uaz — DISMISSED** [review]

File: az_quickstart.install [hunk 4]
```diff
@@ -195,3 +195,10 @@ function az_quickstart_update_1020802() {
 function az_quickstart_update_1020803() {
   \Drupal::service('module_installer')->uninstall(['media_library_theme_reset']);
 }
+
+/**
+ * Enable ckeditor_indentblock module by default.
+ */
+function az_quickstart_update_1020901() {
```
Comment (joeparsons) [inline az_quickstart.install:202]: If this is 2.10.x only I think we need a different number here.
```suggestion
function az_quickstart_update_1021001() {
```

**Review by joeparsons — DISMISSED** [review]

File: az_quickstart.install [hunk 5]
```diff
@@ -195,3 +195,10 @@ function az_quickstart_update_1020802() {
 function az_quickstart_update_1020803() {
   \Drupal::service('module_installer')->uninstall(['media_library_theme_reset']);
 }
+
+/**
+ * Enable ckeditor_indentblock module by default.
+ */
+function az_quickstart_update_1021001() {
+  \Drupal::service('module_installer')->install(['ckeditor_indentblock']);
+}
```
Comment (joeparsons) [inline az_quickstart.install:204]: Should we move this to `az_paragraphs.install` since the module dependency is being added there?

Comment (trackleft) [reply az_quickstart.install:204]: I like that idea

**Review by natebyerley — APPROVED** [review]
I tested this feature.

- Works as expected
- Displays well on mobile
- Works within tables

**Review by joeparsons — APPROVED** [review]

**Review by bberndt-uaz — APPROVED** [review]

**Review by tadean — APPROVED** [review]

**Pull request merged**